src/serialization/read/mod.rs

```diff
@@ -88,6 +88,9 @@ pub fn page_to_array(page: CompressedPage, descriptor: &ColumnDescriptor) -> Res
                 PhysicalType::Double => {
                     Ok(Array::Float64(primitive::page_to_vec(&page, descriptor)?))
                 }
+                PhysicalType::ByteArray => {
```

Comment:
Could we add tests to this?  We generate a parquet file from pyarrow with Binary values, and have the corresponding result written in `fn pyarrow_optional`, which we can run against using `test_pyarrow_integration`. 

It would be good to verify that we can read that file.